source/server/server.cc

```diff
@@ -165,30 +166,26 @@ void InstanceImpl::failHealthcheck(bool fail) {
 }
 
 MetricSnapshotImpl::MetricSnapshotImpl(Stats::Store& store, TimeSource& time_source) {
-  snapped_counters_ = store.counters();
-  counters_.reserve(snapped_counters_.size());
-  for (const auto& counter : snapped_counters_) {
-    counters_.push_back({counter->latch(), *counter});
-  }
+  store.forEachSinkedCounter([this](std::size_t size) mutable { counters_.reserve(size); },
+                             [this](Stats::Counter& counter) mutable {
+                               counters_.push_back({counter.latch(), counter});
```

Comment:
I think we need to plumb this change more deeply to actually provide a benefit in practice. I would argue that makes sense in this PR. WDYT?

Reply:
I am not sure I follow :-(  Could you please elaborate?

Reply:
I think you're all set based on our IM conversation. resolving.